### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by Peter-Dumbari**
## In this PR:
- I created a new branch called `query_and_update`
- I pulled the code from my `development` branch.
- I added some modifications on the project files 
- I created a Pull Request 
- Below are the required screenshots 👇 

Screen Shot 2023-07-19 at 11 41 25 AM

Screen Shot 2023-07-19 at 11 53 55 AM

Screen Shot 2023-07-19 at 1 08 08 PM

Screen Shot 2023-07-19 at 1 20 38 PM

Screen Shot 2023-07-19 at 1 20 54 PM

Screen Shot 2023-07-19 at 5 54 28 PM

Screen Shot 2023-07-19 at 6 03 45 PM

Screen Shot 2023-07-19 at 6 04 55 PM

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -3,3 +3,13 @@ INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts)
 INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES('Gabumon', '2018-11-15', 8, true, 2);
 INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES('Pikachu', '2021-01-07', 15.04, 'false', 1);
 INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES ('Devimon', '2017-05-12', 11, 'true', 5);
+
+
+INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES ('Charmander', '2020-02-08', -11, false, 0);
+INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES ('Plantmon', '2021-11-15', -5.7, true, 2);
+INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES ('Squirtle', '2021-01-07', -12.13, false, 3);
```
Comment (tugsem) [inline data.sql]: - [ ] Please change the date_of birth column of the animals where animal name is 'Squirtle'. You have set the wrong data. It should be like the following: 

>     INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)VALUES ('Squirtle', '1993/04/02', '3', 'false', '-12.13');

File: data.sql [hunk 2]
```diff
@@ -3,3 +3,13 @@ INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts)
...
+INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES ('Angemon', '2005-06-12', -45, true, 1);
+INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES ('Boarmon', '2005-06-07', 20.4, true, 7);
+INSERT INTO animals(name, date_of_birth, weight_kg, neutered, escape_attempts) VALUES ('Blossom', '1998-10-13', 17.3, true, 3);
```
Comment (tugsem) [inline data.sql]: - [ ] Please change the weight_kg column of the animals where animal name is 'Blossom'. You have set the wrong data. It should be like the following:
 ```sql
    INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
    VALUES ('Blossom', '1998/10/13', '3', 'true', '17.0');
```

File: data.sql [hunk 3]
```diff
@@ -3,3 +3,13 @@ INSERT INTO animals (name, date_of_birth, weight_kg, neutered, escape_attempts)
...
+
```
Comment (tugsem) [inline data.sql:15]: - [ ] Please add one more row as per the requirement: 
image

File: queries.sql [hunk 4]
```diff
@@ -28,4 +28,88 @@ WHERE name <> 'Gabumon';
...
-WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+
+BEGIN TRANSACTION;
+
+UPDATE animals
+SET species = 'unspecified';
+
```
Comment (tugsem) [inline queries.sql]: - [ ] As per the requirement, please add the following commands to make sure the change was undone. 
```sql
   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

File: queries.sql [hunk 5]
```diff
@@ -28,4 +28,88 @@ WHERE name <> 'Gabumon';
...
+ROLLBACK TO SAVEPOINT deleted_data_with_date_2021_01_01;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
```
Comment (tugsem) [inline queries.sql]: - [ ] Please remove these lines as per the requirement. There should be only 'COMMIT' after this query.
image

File: queries.sql [hunk 6]
```diff
@@ -28,4 +28,88 @@ WHERE name <> 'Gabumon';
...
+SELECT neutered, COUNT(*) AS escape_count
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered
+ORDER BY escape_count DESC
+LIMIT 1;
+
+SELECT animals, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY animals;
```
Comment (tugsem) [inline queries.sql]: - [ ] The question is 'What is the minimum and maximum weight of each type of animal?'. Animals should be grouped by their species. Kindly fix this.

File: queries.sql [hunk 7]
```diff
@@ -28,4 +28,88 @@ WHERE name <> 'Gabumon';
...
+
+SELECT animals, AVG(escape_attempts) AS average_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY animals;
```
Comment (tugsem) [inline queries.sql]: - [ ] As the one above, there should also be a 'group by species' statement as per the question: 'What is the average number of escape attempts per animal type of those born between 1990 and 2000?'. Kindly fix this and update the screenshot for your applied changes to queries.sql.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @Peter-Dumbari  👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️species table is added
✔️most data  and queries are added to the database
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @Peter-Dumbari  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Made changes as requested by the previous reviewer :+1: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**